zokrates_core/src/flatten/mod.rs

```diff
@@ -2083,6 +2120,24 @@ impl<'ast, T: Field> Flattener<'ast, T> {
         }
     }
 
+    /// Unwraps a boolean & (and) expression chain to a flat vector of boolean expressions
+    ///
+    /// # Arguments
+    /// * `e` - `BooleanExpression` to be flattened
+    fn unwrap_and_chain(
+        &mut self,
+        e: BooleanExpression<'ast, T>,
+    ) -> Vec<BooleanExpression<'ast, T>> {
+        match e {
+            BooleanExpression::And(box lhs, box rhs) => {
+                let mut inner = self.unwrap_and_chain(lhs);
```

Comment:
Why isn't `rhs` also visited here? Because we construct trees where right children are always leaves?

Reply:
Yeah, according to this https://github.com/Zokrates/ZoKrates/blob/develop/zokrates_core/src/static_analysis/flatten_complex_types.rs#L602

Reply:
Okay, I think flattening shouldn't make this assumption though.